### ISSUE TITLE
Heatmap handling of NaN

### ISSUE THREAD

**Issue by fredcallaway** [JuliaPlots/UnicodePlots.jl#362]
It seems that the current behavior of heatmap is to replace NaN values with a grid of transparent and white cells:

<img width="210" alt="image" src="https://github.com/JuliaPlots/UnicodePlots.jl/assets/7883156/10d6eee5-1582-4cdd-a74a-6307098e681f">

I think it would make much more sense if NaN mapped to a consistent color, probably transparent.

**Comment by t-bltg** [comment]
Actually, a row consist of [half blocks characters](https://github.com/JuliaPlots/UnicodePlots.jl/blob/bd10d0714bf9e04d1dd7b021c23661add24c353a/src/common.jl#L229) at the bottom, with a background color for the remaining upper part.

**Comment by fredcallaway** [comment]
Hmm, well then half blocks must be used throughout because the NaN cells are the same size as the other ones (you can see that you get one row for each of the 8 rows in the matrix above).

Implementation aside, the question is: should NaN be represented by a single color everywhere or a different color depending on what row you're in?

**Comment by t-bltg** [comment]
I agree that we should probably switch to no coloring for `NaN`s, to be consistent with other plotting libraries.
